File: 2414100138-张秀华/第一周作业/2414100138-张秀华-第一周代码/app/src/main/java/com/example/music/MainActivity.java
package com.example.music;

import android.Manifest;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.content.res.AssetFileDescriptor;
import android.media.MediaPlayer;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;

import java.io.IOException;

public class MainActivity extends AppCompatActivity {


    private static final int REQUEST_CODE_PERMISSION = 1001;
    private Button bt_startMusic, bt_stopMusic;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        bt_startMusic = findViewById(R.id.bt_startMusic);
        bt_stopMusic = findViewById(R.id.bt_stopMusic);

        //播放音乐
        bt_startMusic.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                Intent intent = new Intent(MainActivity.this, MusicService.class);
                intent.setAction("START");
                intent.putExtra("content", "我的音乐正在后台播放");
                startService(intent);
                Toast.makeText(MainActivity.this, "开始播放音乐啦~", Toast.LENGTH_SHORT).show();

            }
        });

        //停止音乐
        bt_stopMusic.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, MusicService.class);
                intent.setAction("STOP");
                stopService(intent);
                Toast.makeText(MainActivity.this, "音乐暂停啦~", Toast.LENGTH_SHORT).show();
            }
        });

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
            checkNotificationPermission();
        }
    }

    //检查通知权限
    private void checkNotificationPermission() {
        if (ContextCompat.checkSelfPermission(this, Manifest.permission.POST_NOTIFICATIONS)
                != PackageManager.PERMISSION_GRANTED) {

            ActivityCompat.requestPermissions(this,
                    new String[]{Manifest.permission.POST_NOTIFICATIONS},
                    REQUEST_CODE_PERMISSION);
        }
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        if (requestCode == REQUEST_CODE_PERMISSION) {
            if (!(grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED)) {
                Toast.makeText(MainActivity.this, "请在设置中开启通知权限，否则服务无法运行", Toast.LENGTH_LONG).show();
            }
        }
    }

}

File: 2414100138-张秀华/第十周作业/2414100138-张秀华-第十周代码/Music_Kotlin/app/src/main/java/com/example/music_kotlin/MainActivity.kt
package com.example.music_kotlin

import android.Manifest
import android.content.Intent
import android.content.pm.PackageManager
import android.os.Build
import android.os.Bundle
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat

class MainActivity : AppCompatActivity() {

    private val REQUEST_CODE_PERMISSION = 1001
    private lateinit var bt_startMusic: Button
    private lateinit var bt_stopMusic: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        bt_startMusic = findViewById(R.id.bt_startMusic)
        bt_stopMusic = findViewById(R.id.bt_stopMusic)

        // 播放音乐
        bt_startMusic.setOnClickListener {
            val intent = Intent(this@MainActivity, MusicService::class.java).apply {
                action = "START"
                putExtra("content", "我的音乐正在后台播放")
            }
            startService(intent)
            Toast.makeText(this@MainActivity, "开始播放音乐啦~", Toast.LENGTH_SHORT).show()
        }

        // 停止音乐
        bt_stopMusic.setOnClickListener {
            val intent = Intent(this@MainActivity, MusicService::class.java).apply {
                action = "STOP"
            }
            stopService(intent)
            Toast.makeText(this@MainActivity, "音乐暂停啦~", Toast.LENGTH_SHORT).show()
        }

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
            checkNotificationPermission()
        }
    }

    // 检查通知权限
    private fun checkNotificationPermission() {
        if (ContextCompat.checkSelfPermission(this, android.Manifest.permission.POST_NOTIFICATIONS) != PackageManager.PERMISSION_GRANTED) {
            ActivityCompat.requestPermissions(this, arrayOf(Manifest.permission.POST_NOTIFICATIONS), REQUEST_CODE_PERMISSION)
        }
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        if (requestCode == REQUEST_CODE_PERMISSION) {
            if (grantResults.isNotEmpty() && grantResults[0] != PackageManager.PERMISSION_GRANTED) {
                Toast.makeText(this@MainActivity, "请在设置中开启通知权限，否则服务无法运行", Toast.LENGTH_LONG).show()
            }
        }
    }
}